### ISSUE TITLE
Use `polkit` rather then `sudo`

### ISSUE THREAD

**Issue by zer0-x** [korcankaraokcu/PINCE#177]
When looking in some Linux distributions we can see that there are very few packages depending in `sudo`, while most of them especially GUI ones depends on `polkit` for running things as super user or another user.

In ArchLinux we have **19** packages depending in `sudo` while **89** depedning in `polkit`.
> https://archlinux.org/packages/core/x86_64/sudo/
>
> https://archlinux.org/packages/extra/x86_64/polkit/

In Gentoo we have **25** packages depending in `sudo` while **115** depedning in `polkit`.
> https://packages.gentoo.org/packages/app-admin/sudo/reverse-dependencies
>
>https://packages.gentoo.org/packages/sys-auth/polkit/reverse-dependencies

---

The main reasons are:
1. We have more control on privileges using `polkit` than `sudo`. We can restrict privileges to our needs, while using `sudo` we will get the privileges for every thing without restrictions.
2. Inside a desktop environment `polkit` is integrated to show a GUI prompt to enter your password, while `sudo` require a terminal to enter your password. (We will have the ability to run `PINCE` as a normal desktop application)

---
Examples of applications using `polkit`:
- [GameConqueror](https://github.com/scanmem/scanmem) (Memory scanner like PINCE)
- [GParted](https://en.wikipedia.org/wiki/GParted)
- [Flatpak](https://flatpak.org/)
- [Hyprland](https://hyprland.org/)

**Comment by korcankaraokcu** [comment]
Yeah, polkit was mentioned in the past, it could be a nice addition. I'm not working actively on pince right now but I can try to implement it when I come back. Leaving the issue open for now

**Comment by authorisation** [comment]
I was having issues with gdb not being to initialize, turns out this was because I do not have sudo installed. Any reason why there is no exception printed for this in the terminal? Took me some fiddling to figure out.

**Comment by korcankaraokcu** [comment]
> Any reason why there is no exception printed for this in the terminal?

All of distros we support use sudo that's why. You also don't need sudo for the AppImage binary, that "use sudo" message at the startup is just a recommendation since that's what most of the distros use. I'm not against the sudo alternatives but I just don't see it as important as the other features or improvements. I usually spend my free time on other issues. So, any PRs are welcome on this issue

**Comment by Twig6943** [comment]
Any updates on this (this is the only program I run through terminal)

**Comment by brkzlr** [comment]
No updates. If there are any, there would be comments or issue would be closed.

**Comment by korcankaraokcu** [comment]
@Twig6943 Please stop spamming "Any updates?" to issues unless you have something useful to say, or even better, a proper PR

**Comment by Twig6943** [comment]
> @Twig6943 Please stop spamming "Any updates?" to issues unless you have something useful to say, or even better, a proper PR

sorry, won't happen again

**Comment by brkzlr** [comment]
We also use sudo for gdb internally through pexpect (inside debugcore) so it's not just simply a matter of replacing the UID check.

The first one is done in PINCE.sh if you want to tackle it.

**Comment by brkzlr** [comment]
Possibly, I didn't look into it much as we were busy with other more important features.

Give it a try and shoot us a PR if you want to. Just make sure it doesn't break gdb functionality or the keyboard package which also requires root privileges to my knowledge.

Make sure to also modify the AppImage packager that is in `ci/package.sh` as that also uses sudo and will need polkit changes.

There might be a couple of other places that sudo is required but I'm not 100% sure.

**Comment by brkzlr** [comment]
Yeah, that's what I imagined the issues would be. It's not really an easy task, given the way PINCE works, especially with gdb.

This was kinda placed on the backburner for a bit given the low priority and it might be made easier when we tackle the more important stuff, such as replacing gdb with lldb.
If we'll go ahead with this move, we could convert everything into a library and not have to spawn subprocesses anymore which would help a lot with the polkit transition.

**Comment by brkzlr** [comment]
What are you exactly referring to when saying "cosmetic"?

**Comment by brkzlr** [comment]
We will need to preserve themes sadly, it won't be acceptable to lose any functionality that is currently present with sudo.

We'll also have to make sure that PINCE still uses the user's local `.config` folder where we store some settings. Without the environment preservation we do with sudo, those would be written to root's home (and also ignore user's own DE settings such as themes and such).
